moto/route53/models.py

```diff
@@ -277,18 +277,15 @@ def get_all_hosted_zones(self):
         return self.zones.values()
 
     def get_hosted_zone(self, id_):
-        return self.zones.get(id_)
+        return self.zones.get(id_.lstrip("/hostedzone/"))
```

Comment:
Ah, of course. I literally just bumped into this earlier today too, and hadn't learned my lesson.